src/main/java/nz/cri/gns/NSHM/opensha/ruptures/NSHMRuptureSetBuilder.java

```diff
@@ -0,0 +1,265 @@
+			System.out.println("Fault model now has "+fsd.size()+" fault sections");
+		}
+		
+		// build the subsections
+		FaultSectionList subSections = new FaultSectionList(fsd);
+		for (FaultSection parentSect : fsd) {
+			double ddw = parentSect.getOrigDownDipWidth();
+			double maxSectLength = ddw*maxSubSectionLength;
+			System.out.println("Get subSections in "+parentSect.getName());
+			// the "2" here sets a minimum number of sub sections
+			List<? extends FaultSection> newSubSects = parentSect.getSubSectionsList(maxSectLength, subSections.getSafeId(), 2);
+			subSections.addAll(newSubSects);
+			System.out.println("Produced "+newSubSects.size()+" subSections in "+parentSect.getName());
+		}		
+		System.out.println(subSections.size()+" Sub Sections");
+	
+		// @voj is this redundant now we're using FaultSectionList??
+		for (int s=0; s<subSections.size(); s++)
+			Preconditions.checkState(subSections.get(s).getSectionId() == s,
+				"section at index %s has ID %s", s, subSections.get(s).getSectionId());
+			
+		SectionDistanceAzimuthCalculator distAzCalc = new SectionDistanceAzimuthCalculator(subSections);
+		JumpAzimuthChangeFilter.AzimuthCalc azimuthCalc = new JumpAzimuthChangeFilter.SimpleAzimuthCalc(distAzCalc);
+		
+		// connection strategy: parent faults connect at closest point, and only when dist <=5 km
+		ClusterConnectionStrategy connectionStrategy = new DistCutoffClosestSectClusterConnectionStrategy(subSections, distAzCalc, maxDistance);
+		System.out.println("Built connectionStrategy");
+		
+		int maxNumSplays = 0; // don't allow any splays
+	
+		Predicate<FaultSubsectionCluster> pFilter = new SubSectionParentFilter().makeParentIdFilter(faultIdIn);
+		PlausibilityConfiguration config =
+				PlausibilityConfiguration.builder(connectionStrategy, distAzCalc)
+						.maxSplays(maxNumSplays)
+						.add(new SubSectionParentFilter(pFilter))
+						.add(new JumpAzimuthChangeFilter(azimuthCalc, 60f))
+						.add(new TotalAzimuthChangeFilter(azimuthCalc, 60f, true, true))
+						.add(new CumulativeAzimuthChangeFilter(azimuthCalc, 560f))
+						.add(new MinSectsPerParentFilter(minSubSectsPerParent, true, true, connectionStrategy))
+						.build();
+		System.out.println("Built PlausibilityConfiguration");
+		
+		// Builder can now proceed using the clusters and all the filters...
+		ClusterRuptureBuilder builder = new ClusterRuptureBuilder(config);
+		System.out.println("initialised ClusterRuptureBuilder");
+		
+		ClusterPermutationStrategy permutationStrategy = createPermutationStrategy(permutationStrategyClass);
+				
+		List<ClusterRupture> ruptures = builder.build(permutationStrategy, numThreads);
+		System.out.println("Built "+ruptures.size()+" total ruptures");
+		
+		NSHMSlipEnabledRuptureSet rupSet = new NSHMSlipEnabledRuptureSet(ruptures, subSections,
+				ScalingRelationships.SHAW_2009_MOD, SlipAlongRuptureModels.UNIFORM);
+		rupSet.setPlausibilityConfiguration(config);
+		return (SlipAlongRuptureModelRupSet) rupSet;
```

Comment:
Does Java complain if you don't cast here? I think it should work without it.

Reply:
you're right thanks!